### PR TITLE
Avoid NPE for annotation arg

### LINKED ISSUE #12538: String Interpolation in annotation parameter causes NullPointerException
## reproduction steps

using Scala 2.13.8 (OpenJDK 64-Bit Server VM, Java 11.0.13),

```scala
scala> @java.lang.Deprecated(since = s"test") var i = 4
```

## problem
When I try to use string interpolation for annotation parameter it causes following compiler exception.

`java.lang.NullPointerException
	at scala.tools.nsc.typechecker.Typers$Typer.isDefaultArg$1(Typers.scala:3963)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedAnnotation$11(Typers.scala:3969)
	at scala.tools.nsc.typechecker.Typers$Typer.typedAnnotation(Typers.scala:4055)
	at scala.tools.nsc.typechecker.Namers$Namer.computeInfo$1(Namers.scala:1898)
	at scala.tools.nsc.typechecker.Namers$Namer.$anonfun$annotSig$6(Namers.scala:1912)
	at scala.reflect.internal.AnnotationInfos$LazyAnnotationInfo.forcedInfo$lzycompute(AnnotationInfos.scala:162)
	at scala.reflect.internal.AnnotationInfos$LazyAnnotationInfo.forcedInfo(AnnotationInfos.scala:162)
	at scala.reflect.internal.AnnotationInfos$LazyAnnotationInfo.completeInfo(AnnotationInfos.scala:175)
	at scala.tools.nsc.typechecker.Typers$Typer.typedValDefImpl(Typers.scala:2127)
	at scala.tools.nsc.typechecker.Typers$Typer.typedValDef(Typers.scala:2111)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5949)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:6041)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:6119)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$8(Typers.scala:3410)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3410)
	at scala.tools.nsc.typechecker.Typers$Typer.typedTemplate(Typers.scala:2064)
	at scala.tools.nsc.typechecker.Typers$Typer.typedClassDef(Typers.scala:1895)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5951)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:6041)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:6119)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$8(Typers.scala:3410)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3410)
	at scala.tools.nsc.typechecker.Typers$Typer.typedTemplate(Typers.scala:2064)
	at scala.tools.nsc.typechecker.Typers$Typer.typedClassDef(Typers.scala:1895)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5951)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:6041)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStat$1(Typers.scala:6119)
	at scala.tools.nsc.typechecker.Typers$Typer.$anonfun$typedStats$8(Typers.scala:3410)
	at scala.tools.nsc.typechecker.Typers$Typer.typedStats(Typers.scala:3410)
	at scala.tools.nsc.typechecker.Typers$Typer.typedPackageDef$1(Typers.scala:5634)
	at scala.tools.nsc.typechecker.Typers$Typer.typed1(Typers.scala:5954)
	at scala.tools.nsc.typechecker.Typers$Typer.typed(Typers.scala:6041)
	at scala.tools.nsc.typechecker.Analyzer$typerFactory$TyperPhase.apply(Analyzer.scala:117)
	at scala.tools.nsc.Global$GlobalPhase.applyPhase(Global.scala:459)
	at scala.tools.nsc.typechecker.Analyzer$typerFactory$TyperPhase.run(Analyzer.scala:106)
	at scala.tools.nsc.Global$Run.compileUnitsInternal(Global.scala:1524)
	at scala.tools.nsc.Global$Run.compileUnits(Global.scala:1508)
	at scala.tools.nsc.interpreter.IMain$ReadEvalPrint.compile(IMain.scala:744)
	at scala.tools.nsc.interpreter.IMain$Request.compile(IMain.scala:967)
	at scala.tools.nsc.interpreter.IMain.compile(IMain.scala:533)
	at scala.tools.nsc.interpreter.IMain.doInterpret(IMain.scala:519)
	at scala.tools.nsc.interpreter.IMain.interpret(IMain.scala:503)
	at scala.tools.nsc.interpreter.IMain.interpret(IMain.scala:501)
	at scala.tools.nsc.interpreter.shell.ILoop.loop$1(ILoop.scala:878)
	at scala.tools.nsc.interpreter.shell.ILoop.interpretStartingWith(ILoop.scala:906)
	at scala.tools.nsc.interpreter.shell.ILoop.command(ILoop.scala:433)
	at scala.tools.nsc.interpreter.shell.ILoop.processLine(ILoop.scala:440)
	at scala.tools.nsc.interpreter.shell.ILoop.loop(ILoop.scala:458)
	at scala.tools.nsc.interpreter.shell.ILoop.run(ILoop.scala:968)
	at scala.tools.nsc.MainGenericRunner.runTarget$1(MainGenericRunner.scala:87)
	at scala.tools.nsc.MainGenericRunner.run$1(MainGenericRunner.scala:91)
	at scala.tools.nsc.MainGenericRunner.process(MainGenericRunner.scala:103)
	at scala.tools.nsc.MainGenericRunner$.main(MainGenericRunner.scala:108)
	at scala.tools.nsc.MainGenericRunner.main(MainGenericRunner.scala)
`

### PR DESCRIPTION
**Pull request by som-snytt**
Fixes scala/bug#12538

Needs more to handle constant result of interpolation.

### CODE REVIEW

**Comment by som-snytt** [comment]
Just noticed I forgot the check file. Recently I saw on a 2018 comment, I suggested
```
partest --check-check
```
to ensure check files were committed.

**Comment by som-snytt** [comment]
Making the fast-tracked interpolators whitebox suffices to allow constant folding, presumably `"hello, world": String` becomes `"hello, world": "hello, world"` as it ought to be.

I haven't looked into it too closely yet. "Too closely" implies that one ought to remain at a safe distance.

**Comment by som-snytt** [comment]
oh, you can't name Java args on obsolete JVMs!

```
!!    1 - neg/t12538.scala                          [output differs]
% diff /home/jenkins/workspace/scala-2.13.x-validate-main/test/files/neg/t12538.check /home/jenkins/workspace/scala-2.13.x-validate-main/test/files/neg/t12538-neg.log
@@ -1,4 +1,4 @@
-t12538.scala:3: error: annotation argument needs to be a constant; found: ("test": String)
+t12538.scala:3: error: unknown annotation argument name: since
   @java.lang.Deprecated(since = s"test") var i = 4
-                                ^
+                              ^
 1 error
```
Maybe I can craft a test that runs conditionally.

**Review by SethTisue — APPROVED** [review]

**Comment by som-snytt** [comment]
I couldn't remember where I stashed the whitebox interpolators, I needed them recently.

File: test/files/run/t12538.scala [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+object Test extends DirectTest {
+  def code =
+    """
+      |class C {
+      |  @java.lang.Deprecated(since = s"test") var i = 4
+      |}
+    """.stripMargin
+
+  def show() = testUnderJavaAtLeast("17")(assert(compile()))
```
Comment (lrytz) [inline test/files/run/t12538.scala]: There's also

> A `// javaVersion <N[+| - M]>` comment makes partest skip the test if the java version is outside the requested range

https://github.com/scala/scala/blob/2.13.x/CONTRIBUTING.md#partest

**Pull request merged**